### PR TITLE
feat(FEC-8394): move rates to engine

### PR DESCRIPTION
**Pull request by itaykinnrot**
### Description of the Changes
remove the hard coded rates and move them to the player engine + support different rates for live and vod (remove this logic from the ui...)

### CheckLists

- [X] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [X] test are passing in local environment 
- [X] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -42,19 +42,7 @@ import PKError from './error/error'
...
- * @type {Array<number>}
- * @const
- */
-const PLAYBACK_RATES = [0.5, 1, 2, 4];
 
-/**
- * The player default playback rate.
- * @type {number}
- * @const
- */
```
Comment (odedhutzler) [inline src/player.js]: why delete that const?

File: src/engines/html5/html5.js [hunk 2]
```diff
@@ -149,6 +149,20 @@ export default class Html5 extends FakeEventTarget implements IEngine {
...
+   * The player playback rates.
+   * @type {Array<number>}
+   */
+   PLAYBACK_RATES = [0.5, 1, 2, 4];
+
+   /**
+   * The player playback rates.
+   * @type {Array<number>}
+   */
+   LIVE_PLAYBACK_RATES = [1];
```
Comment (yairans) [inline src/engines/html5/html5.js]: should be `static const`

Comment (yairans) [reply src/engines/html5/html5.js]: also fix the jsdoc. also above

File: src/engines/html5/html5.js [hunk 3]
```diff
@@ -840,6 +854,14 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     return this._el.getAttribute('crossorigin');
   }
 
+  /**
+   * get the playback rates
+   * @return {number[]} - plyback rates
```
Comment (yairans) [inline src/engines/html5/html5.js]: `Array<number>`. typo - `plyback `

File: src/engines/html5/html5.js [hunk 4]
```diff
@@ -840,6 +854,14 @@ export default class Html5 extends FakeEventTarget implements IEngine {
...
+   */
+  get playbackRates(): Array<number> {
+    return this.isLive()? this.LIVE_PLAYBACK_RATES : this.PLAYBACK_RATES;
```
Comment (yairans) [inline src/engines/html5/html5.js]: should be static.

File: src/engines/html5/html5.js (hunk 2 above)
Comment (OrenMe) [reply src/engines/html5/html5.js]: @itaykinnrot why differentiate live and vod?

File: src/engines/html5/html5.js (hunk 4 above)
Comment (OrenMe) [reply src/engines/html5/html5.js]: again, why differentiate?

File: src/engines/html5/html5.js (hunk 2 above)
Comment (itaykinnrot) [reply src/engines/html5/html5.js]: @OrenMe  to keep the same behaviour we have today - i removed it from the ui

Comment (OrenMe) [reply src/engines/html5/html5.js]: The reason it was in the UI is that we don’t want to limit the API of player with BL.
The UI in this case is opinionated and can make assumptions but from playback perspective having rate higher or lower than 1 for live is also possible(DVR playback).
Please keep this logic in UI.

**Pull request merged**